artpar 2nd attempt

artpar opened the pull request:
#### Check List:
- [x] Tests pass (`./test.sh <username>`)
- [x] All formatting changes by the build are committed
- [x] Output matches that of calculate_average_baseline.sh

* Execution time: 00:5.127
* Execution time of reference implementation: 3:08.77

```
[base]                   183.28s user 5.63s system 100% cpu 3:08.77 total
[memory mapped+parallel] 167.33s user 5.04s system 630% cpu 27.321 total
[above+vector]           157.12s user 3.43s system 670% cpu 23.958 total
[update]                 78.74s user  6.10s system 680% cpu 12.459 total
[b7385ed]                65.03s user  5.68s system 664% cpu 10.634 total
[dda0f36]                61.29s user  5.36s system 727% cpu 9.163 total
(^ previous pr)
[4e6dc45]                35.99s user  5.96s system 680% cpu 6.168 total
[cbdb861]                27.58s user  5.66s system 648% cpu 5.127 total
```

<!--
Thanks for your submission. Please go through the checklist above before submitting your pull request.
Use [x] to mark that the item has been completed.

Please submit only submissions that are taking less than 1 minute.

Please make sure that you have followed the defined rules (https://github.com/gunnarmorling/1brc?tab=readme-ov-file#rules-and-limits)
-->

gunnarmorling commented:
There are test failures. Please run `./test.sh artpar` and make sure no differences are reported.

artpar commented:
I have verified that all tests pass now (there are considerable changes since the last commit, i tried to reproduce test failures on that commit but didn't get any fail). Also verified the other two check marks again as well

gunnarmorling commented:
00:08.398 now.
